src/core/qgsvectorlayer.h

```diff
@@ -953,10 +953,18 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer
     virtual QString subsetString();
 
     /**
-     * Query the provider for features specified in request.
+     * Query the layer for features specified in request.
      */
     QgsFeatureIterator getFeatures( const QgsFeatureRequest& request = QgsFeatureRequest() );
 
+    /**
+     * Query the layer for features that match the given filter expressions
+     * @param filterExpression An expression that features on this layer must match to be returned
+     *                         by this iterator
+     * @return A feature iterator
+     */
+    QgsFeatureIterator getFeatures( const QString& filterExpression );
+
```

Comment:
Couldn't we use keywords for that in python?
```
layer.getFeatures( expression='a>2' )
layer.getFeatures( filterRect=QgsRectangle( 10, 10, 200, 200 ) )
```

I think having some shorthands for this would be great.